### PR TITLE
Fix CI, add Python 3.10

### PR DESCRIPTION
**Pull request by johnnynunez**
- [X] Python 3.6 & 3.7 deleted
- [X] Python 3.10 & 3.11 added
- [X] Flann is required
- [X] Maintain macOS 11 only. 
- [X] Pybind11 compatible with Python 3.11

### CODE REVIEW

**Comment by johnnynunez** [comment]
@Phil26AT now tests on macos are working...
WIP linux.. I have the error 
```
CMake Error at CMakeLists.txt:5 (find_package):
-- Configuring incomplete, errors occurred!
See also "/io/build/temp.linux-x86_64-cpython-38/CMakeFiles/CMakeOutput.log".
See also "/io/build/temp.linux-x86_64-cpython-38/CMakeFiles/CMakeError.log".
  By not providing "FindFLANN.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "FLANN", but
  CMake did not find one.

  Could not find a package configuration file provided by "FLANN" with any of
  the following names:

    FLANNConfig.cmake
    flann-config.cmake

  Add the installation prefix of "FLANN" to CMAKE_PREFIX_PATH or set
  "FLANN_DIR" to a directory containing one of the above files.  If "FLANN"
  provides a separate development package or SDK, be sure it has been
  installed.

```

**Comment by Skydes** [comment]
Thanks for looking into this. I'm getting too many notifications so muting this PR for now - give us a heads up once it is ready for a full review.

File: .github/workflows/build-new.yml [hunk 1]
```diff
@@ -7,76 +7,49 @@ on:
   pull_request:
     types: [ assigned, opened, synchronize, reopened ]
   release:
-    types: [published, edited]
+    types: [ published, edited ]
   workflow_dispatch:
 
 jobs:
-  linux-build:
```
Comment (Skydes) [inline .github/workflows/build-new.yml:14]: Please re-enable the Linux build.

File: .github/workflows/build-new.yml [hunk 2]
```diff
@@ -7,76 +7,49 @@ on:
...
-        path: wheelhouse/pycolmap-*-${{ matrix.pyversion }}-manylinux2014_x86_64.whl
-
   mac-build:
     name: Wrapper macOS Build
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [macos-11, macos-10.15]
-        pyversion: ["python@3.7", "python@3.8", "python@3.9"]
+        os: [ macos-latest ]
```
Comment (Skydes) [inline .github/workflows/build-new.yml:19]: Please re-enable MacOS 11.

**Comment by johnnynunez** [comment]
> Thanks for looking into this. I'm getting too many notifications so muting this PR for now - give us a heads up once it is ready for a full review.

oh sorry so much, I didn't know that!

File: .github/workflows/build-new.yml [hunk 3]
```diff
@@ -7,76 +7,49 @@ on:
...
+        pyversion: [ "python@3.8", "python@3.9", "python@3.10", "python@3.11" ]
```
Comment (Skydes) [inline .github/workflows/build-new.yml:20]: Please re-add 3.7 or give a valid reason for removal.

File: .gitignore [hunk 4]
```diff
@@ -2,3 +2,4 @@
 *.so
 *.egg-info/
 build/
+.idea/
```
Comment (Skydes) [inline .gitignore:5]: Remove.

File: CMakeLists.txt [hunk 5]
```diff
@@ -2,7 +2,7 @@ cmake_minimum_required(VERSION 3.6)
 project(PyCOLMAP)
 
 find_package(COLMAP REQUIRED)
-include_directories(${COLMAP_INCLUDE_DIRS} ${PROJECT_SOURCE_DIR})
+include_directories(${COLMAP_INCLUDE_DIRS} ${PROJECT_SOURCE_DIR} ${COLMAP_EXTERNAL_LIBRARIES})
```
Comment (Skydes) [inline CMakeLists.txt:5]: Why?

File: package/build-wheels-linux-new.sh [hunk 6]
```diff
@@ -0,0 +1,116 @@
+#!/bin/bash
```
Comment (Skydes) [inline package/build-wheels-linux-new.sh:1]: Remove?

File: setup.py [hunk 7]
```diff
@@ -82,7 +82,7 @@ def build_extension(self, ext):
 
 setup(
     name='pycolmap',
-    version='0.4.0',
+    version='0.5.0',
```
Comment (Skydes) [inline setup.py:85]: Please revert this change.

File: .github/workflows/build-new.yml (hunk 3 above)
Comment (johnnynunez) [reply .github/workflows/build-new.yml:48]: Because anaconda remove from his channel and colab etc upgrade to 3.8

**Comment by johnnynunez** [comment]
@Skydes First of all, apologies for the commits and notifications. It's difficult to work only with [act](https://github.com/nektos/act).
For my part it would be, I think. I hope it will be the final one.

**Comment by johnnynunez** [comment]
@Skydes First of all, apologies for the commits and notifications. It's difficult to work only with [act](https://github.com/nektos/act).
For my part it work done, I think. I hope it will be the final one.

**Comment by johnnynunez** [comment]
@Skydes How is it going?

**Comment by mihaidusmanu** [comment]
Looks good to me overall. Let's wait for Paul to take a look when he has a moment too. We will merge some other PRs that were stuck due to broken CI and then bump up version to 0.5 too.

**Comment by Skydes** [comment]
Let's delay the pybind upgrade because we'd need to simultaneously upgrade it in pyceres and pixsfm as well. This forces us to push back Python 3.11 support (though I don't think this is a big deal for now).

**Review by Skydes — APPROVED** [review]
